webui/react/src/utils/data.ts

```diff
@@ -19,14 +20,25 @@ export const isSyncFunction = (fn: unknown): boolean => {
   return isFunction(fn) && !isAsyncFunction(fn);
 };
 
-/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
 export const clone = (data: any, deep = true): any => {
   if (isPrimitive(data)) return data;
   if (isMap(data)) return new Map(data);
   if (isSet(data)) return new Set(data);
   return deep ? JSON.parse(JSON.stringify(data)) : { ...data };
 };
 
+export const getOrElse = <T = any>(
+  obj: Record<string, any>,
+  path: string,
+  fallback?: T,
+): T | undefined => {
+  if (!obj) return fallback;
+
+  // Reassigns to obj[key] on each array.every iteration
+  let value = obj;
+  return path.split('.').every(key => ((value = value[key]) !== undefined)) ? value as T : fallback;
```

Comment:
this usage would be simple with any but wouldn't guarantee that both key1 and key2 would have same types. . wdy think about removing `= any` fallback? could it work as well?

Reply:
there wasn't an easy way to avoid `any` because the intermediate values would really be a `Record` type of unknown/any key-values. So this function will attempt to traverse as far as it can get and whenever it runs into an issue, it will just send back the optional fallback.

Reply:
discussed with @hamidzr : suggest: remove `= any` from `<T = any>` to encourage use of generic type for `getOrElse`